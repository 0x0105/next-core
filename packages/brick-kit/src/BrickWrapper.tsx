import React from "react";
import i18n from "i18next";
import { ConfigProvider } from "antd";
import zhCN from "antd/es/locale/zh_CN";
import enUS from "antd/es/locale/en_US";

import { ErrorBoundary } from "./ErrorBoundary";
import { renderEasyopsEmpty } from "./EasyopsEmpty";

interface BrickWrapperProps {
  children?: React.ReactElement;
}

/**
 * 构件的 React 组件包装器，包含 ErrorBoundary 和 ConfigProvider。
 */
export function BrickWrapper(props: BrickWrapperProps): React.ReactElement {
  const locale =
    i18n.language && i18n.language.split("-")[0] === "en" ? enUS : zhCN;
  return (
    <ErrorBoundary>
      <ConfigProvider
        locale={locale}
        autoInsertSpaceInButton={false}
        renderEmpty={renderEasyopsEmpty}
      >
        {props.children}
      </ConfigProvider>
    </ErrorBoundary>
  );
}
