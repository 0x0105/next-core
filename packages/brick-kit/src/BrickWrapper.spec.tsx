import React from "react";
import { shallow } from "enzyme";
import { BrickWrapper } from "./BrickWrapper";

jest.mock("i18next", () => ({ language: "zh" }));
jest.mock("antd/es/locale/zh_CN", () => "zh_CN");
jest.mock("antd/es/locale/en_US", () => "en_US");

describe("brick wrapper", () => {
  it("should work", () => {
    const wrapper = shallow(
      <BrickWrapper>
        <div>hello, brick-wrapper</div>
      </BrickWrapper>
    );
    expect(wrapper).toMatchSnapshot();
  });
});
