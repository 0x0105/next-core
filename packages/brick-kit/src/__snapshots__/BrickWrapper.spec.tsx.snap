// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`brick wrapper should work 1`] = `
<LegacyErrorBoundary
  i18n={
    Object {
      "language": "zh-CN",
    }
  }
  t={[Function]}
>
  <ConfigProvider
    locale="zh_CN"
  >
    <div>
      hello, brick-wrapper
    </div>
  </ConfigProvider>
</LegacyErrorBoundary>
`;
